# **LIMPIEZA DE DATASETS Y OBTENCIÓN DE DATASET PROCESADO**

In [1]:
import numpy as np
import pandas as pd
pd.set_option('display.max_columns', None)

In [2]:
earthquakes=pd.read_csv(r"C:\Users\isabe\Documents\Thebridge\Machine learning datasets\earthquakeraw.csv")

In [3]:
earthquakes

id  year                   locationName  latitude  longitude  \
0         1 -2150  JORDAN:  BAB-A-DARAA,AL-KARAK    31.100     35.500   
1         2 -2000                 SYRIA:  UGARIT    35.683     35.800   
2         3 -2000               TURKMENISTAN:  W    38.000     58.200   
3         8 -1566       ISRAEL:  ARIHA (JERICHO)    31.500     35.300   
4        11 -1450           ITALY:  LACUS CIMINI    35.500     25.500   
...     ...   ...                            ...       ...        ...   
6600  10923  2025   RUSSIA:  KAMCHATKA PENINSULA    52.512    160.324   
6601  10924  2025              TURKEY: BALIKESIR    39.312     28.069   
6602  10925  2025          EL SALVADOR-GUATEMALA    14.055    -89.883   
6603  10926  2025    INDONESIA:  SULAWESI:  POSO    -1.282    120.727   
6604  10927  2025           CHILE: DRAKE PASSAGE   -60.186    -61.821   

      eqMagnitude  damageAmountOrder  eqMagUnk  publish       country  \
0             7.3                3.0       7.3     True        JORDAN   
1             NaN                NaN       NaN     True         SYRIA   
2             7.1                1.0       NaN     True  TURKMENISTAN   
3             NaN                3.0       NaN     True        ISRAEL   
4             NaN                NaN       NaN     True         ITALY   
...           ...                ...       ...      ...           ...   
6600          8.8                NaN       NaN     True        RUSSIA   
6601          6.1                2.0       NaN     True        TURKEY   
6602          5.7                3.0       NaN     True     GUATEMALA   
6603          5.8                2.0       NaN     True     INDONESIA   
6604          7.5                NaN       NaN     True    ANTARCTICA   

      regionCode  intensity  deathsAmountOrder  tsunamiEventId  \
0            140        NaN                NaN             NaN   
1            140       10.0                3.0             1.0   
2             40       10.0                1.0             NaN   
3            140       10.0                NaN             NaN   
4            130       10.0                NaN             NaN   
...          ...        ...                ...             ...   
6600          50        8.0                NaN          6044.0   
6601         140        NaN                1.0             NaN   
6602         100        NaN                1.0             NaN   
6603         170        NaN                1.0          6046.0   
6604          20        NaN                NaN          6047.0   

      deathsAmountOrderTotal  eqDepth  deaths  housesDestroyedAmountOrder  \
0                        NaN      NaN     NaN                         NaN   
1                        3.0      NaN     NaN                         NaN   
2                        1.0     18.0     1.0                         1.0   
3                        NaN      NaN     NaN                         NaN   
4                        NaN      NaN     NaN                         NaN   
...                      ...      ...     ...                         ...   
6600                     1.0     35.0     NaN                         NaN   
6601                     1.0     10.0     1.0                         1.0   
6602                     1.0     10.0     1.0                         NaN   
6603                     1.0      8.0     1.0                         1.0   
6604                     NaN     10.0     NaN                         NaN   

      eqMagMs  deathsTotal  damageAmountOrderTotal  \
0         NaN          NaN                     NaN   
1         NaN          NaN                     NaN   
2         7.1          1.0                     1.0   
3         NaN          NaN                     NaN   
4         NaN          NaN                     NaN   
...       ...          ...                     ...   
6600      8.0          1.0                     2.0   
6601      NaN          1.0                     2.0   
6602      NaN          1.0                     3.0 

In [4]:
tsunamis=pd.read_csv(r"C:\Users\isabe\Documents\Thebridge\Machine learning datasets\tsunamiraw.csv")

In [5]:
tsunamis

id  year  eventValidity  causeCode  earthquakeEventId  numDeposits  \
0        1 -2000              1        1.0                2.0            0   
1        3 -1610              4        6.0             5877.0           28   
2        4 -1365              1        2.0             9712.0            0   
3        5 -1300              2        0.0                NaN            0   
4        6  -760              2        0.0                NaN            0   
...    ...   ...            ...        ...                ...          ...   
2984  6043  2006             -1        1.0            10922.0            0   
2985  6044  2025              4        1.0            10923.0            0   
2986  6045  2025              4        8.0                NaN            0   
2987  6046  2025              3        1.0            10926.0            0   
2988  6047  2025              4        1.0            10927.0            0   

         country                       locationName  regionCode  latitude  \
0          SYRIA                      SYRIAN COASTS        50.0    35.683   
1         GREECE           THERA ISLAND (SANTORINI)        50.0    36.400   
2          SYRIA                      SYRIAN COASTS        50.0    35.683   
3         TURKEY               IONIAN COASTS, TROAD        50.0    39.960   
4         ISRAEL          ISRAEL AND LEBANON COASTS        50.0       NaN   
...          ...                                ...         ...       ...   
2984      BRAZIL  ARQUIPELAGO SAO PEDRO E SAO PAULO        71.0     1.175   
2985      RUSSIA        KAMCHATKA PENINSULA, RUSSIA        86.0    52.512   
2986         USA                      TRACY ARM, AK        87.0    57.840   
2987   INDONESIA                           SULAWESI        83.0    -1.282   
2988  ANTARCTICA                      DRAKE PASSAGE        71.0   -60.186   

      longitude  numRunups  tsIntensity  deathsAmountOrder  damageAmountOrder  \
0        35.800          1          6.0                3.0                4.0   
1        25.400          2          6.0                3.0                3.0   
2        35.800          1          6.0                NaN                2.0   
3        26.240          0          5.0                NaN                NaN   
4           NaN          0          3.0                NaN                NaN   
...         ...        ...          ...                ...                ...   
2984    -28.065          0          NaN                NaN                NaN   
2985    160.324        332          NaN                1.0                2.0   
2986   -133.060          3          NaN                NaN                1.0   
2987    120.727          1          NaN                NaN                NaN   
2988    -61.821          2          NaN                NaN                NaN   

      deathsAmountOrderTotal  damageAmountOrderTotal  publish  oceanicTsunami  \
0                        3.0                     4.0     True            True   
1                        3.0                     3.0     True            True   
2                        NaN                     3.0     True            True   
3                        NaN                     NaN     True            True   
4                        NaN                     NaN     True            True   
...                      ...                     ...      ...             ...   
2984                     NaN                     1.0     True            True   
2985                     1.0                     2.0     True            True   
2986                     NaN                     1.0     True            True   
2987                     1.0                     2.0     True            True   
2988                     NaN                     NaN     True            True   

      volcanoEventId  maxWaterHeight  eqMagnitude  month  \
0                NaN             NaN          NaN    NaN   
1             1351.0           90.00          NaN    NaN   
2                NaN

## 1.- LIMPIEZA INICIAL

Como tenemos unos datasets con muchísimos registros, voy a eliminar las filas en las que faltan la latitud y la longitud en ambos datasets.

- En el dataset de terremotos, pasamos de 6605 filas a 6544.

Además, en el dataset de tsunamis, eliminaré las filas en las que no aparece el id del terremoto, ya que no se puede relacionar dichas filas con las correspondientes al dataset de terremotos.

- En el dataset de tsunamis pasamos de 2989 filas a 2020 filas.

Para futuras predicciones, tendríamos que realizar los siguientes pasos con los nuevos datasets para utilizarlos en el modelo de predicción final.

In [6]:
earthquakes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6605 entries, 0 to 6604
Data columns (total 49 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   id                               6605 non-null   int64  
 1   year                             6605 non-null   int64  
 2   locationName                     6605 non-null   object 
 3   latitude                         6544 non-null   float64
 4   longitude                        6544 non-null   float64
 5   eqMagnitude                      4808 non-null   float64
 6   damageAmountOrder                4824 non-null   float64
 7   eqMagUnk                         822 non-null    float64
 8   publish                          6605 non-null   bool   
 9   country                          6605 non-null   object 
 10  regionCode                       6605 non-null   int64  
 11  intensity                        3333 non-null   float64
 12  deathsAmountOrder   

In [7]:
earthquakes.isnull().sum()

id                                    0
year                                  0
locationName                          0
latitude                             61
longitude                            61
eqMagnitude                        1797
damageAmountOrder                  1781
eqMagUnk                           5783
publish                               0
country                               0
regionCode                            0
intensity                          3272
deathsAmountOrder                  3859
tsunamiEventId                     4529
deathsAmountOrderTotal             4093
eqDepth                            3039
deaths                             4365
housesDestroyedAmountOrder         4619
eqMagMs                            3527
deathsTotal                        4503
damageAmountOrderTotal             2576
housesDestroyedAmountOrderTotal    4534
month                               412
day                                 569
volcanoEventId                     6525


In [8]:
earthquakes = earthquakes.dropna(subset=["latitude", "longitude"]).reset_index(drop=True)

In [9]:
earthquakes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6544 entries, 0 to 6543
Data columns (total 49 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   id                               6544 non-null   int64  
 1   year                             6544 non-null   int64  
 2   locationName                     6544 non-null   object 
 3   latitude                         6544 non-null   float64
 4   longitude                        6544 non-null   float64
 5   eqMagnitude                      4806 non-null   float64
 6   damageAmountOrder                4800 non-null   float64
 7   eqMagUnk                         822 non-null    float64
 8   publish                          6544 non-null   bool   
 9   country                          6544 non-null   object 
 10  regionCode                       6544 non-null   int64  
 11  intensity                        3321 non-null   float64
 12  deathsAmountOrder   

In [10]:
tsunamis.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2989 entries, 0 to 2988
Data columns (total 52 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   id                               2989 non-null   int64  
 1   year                             2989 non-null   int64  
 2   eventValidity                    2989 non-null   int64  
 3   causeCode                        2986 non-null   float64
 4   earthquakeEventId                2076 non-null   float64
 5   numDeposits                      2989 non-null   int64  
 6   country                          2988 non-null   object 
 7   locationName                     2987 non-null   object 
 8   regionCode                       2988 non-null   float64
 9   latitude                         2599 non-null   float64
 10  longitude                        2598 non-null   float64
 11  numRunups                        2989 non-null   int64  
 12  tsIntensity         

In [11]:
tsunamis = tsunamis.dropna(subset=["latitude", "longitude"]).reset_index(drop=True)

In [12]:
tsunamis = tsunamis.dropna(subset=["earthquakeEventId"]).reset_index(drop=True)

In [13]:
tsunamis.isnull().sum()

id                                    0
year                                  0
eventValidity                         0
causeCode                             1
earthquakeEventId                     0
numDeposits                           0
country                               0
locationName                          1
regionCode                            1
latitude                              0
longitude                             0
numRunups                             0
tsIntensity                        1068
deathsAmountOrder                  1742
damageAmountOrder                  1532
deathsAmountOrderTotal             1301
damageAmountOrderTotal              881
publish                               0
oceanicTsunami                        0
volcanoEventId                     1963
maxWaterHeight                      971
eqMagnitude                         381
month                                64
housesDestroyedAmountOrder         1788
deathsTotal                        1443


## 2.- SOLUCIÓN A LOS DIFERENTES TIPOS DE ESCALAS DE MAGNITUD EN EL DATASET DE TERREMOTOS.

En el dataset de terremotos, existe un problema con las escalas de las magnitudes.

Para todas las escalas, la magnitud es un valor adimensional, pero no es intercambiable entre dichas escalas.

&nbsp;

Las escalas presentes en el dataset son:

* Mw (Moment Magnitude): estándar actual, más consistente para eventos grandes.

* Ms (Surface wave magnitude): basada en ondas superficiales.

* Mb (Body wave magnitude): ondas de cuerpo, útil para distancias largas.

* ML (Local magnitude o Richter original): ondas locales.  

&nbsp;

Algunas de las ecuaciones empíricas de conversión son:

| De → A       | Fórmula aproximada                      | Referencia       |
| ------------ | --------------------------------------- | ---------------- |
| **Ms → Mw**  | Mw ≈ 0.67 × Ms + 2.07                   | Hanks & Kanamori |
| **Mb → Mw**  | Mw ≈ 0.85 × Mb + 1.03                   | Scordilis, 2006  |
| **Ml → Mw**  | Mw ≈ 0.85 × Ml + 1.03                   | Estimación común |
| **Mfa → Mw** | Aproximar igual que Ms (si no hay otra) |                  |


**Nota:** Estas relaciones son **aproximaciones empíricas**, las conversiones pueden no ser precisas.




In [14]:
# Vamos a crear una nueva columna que se llama magnitude_Mw aplicando las fórmulas anteriores a nuestro dataset de terremotos.
# En la nueva columna obtenida, eliminamos las filas en las que el dato falte.

def to_mw(row):
    if not pd.isna(row["eqMagMw"]):
        return row["eqMagMw"]
    elif not pd.isna(row["eqMagMs"]):
        return 0.67 * row["eqMagMs"] + 2.07
    elif not pd.isna(row["eqMagMb"]):
        return 0.85 * row["eqMagMb"] + 1.03
    elif not pd.isna(row["eqMagMl"]):
        return 0.85 * row["eqMagMl"] + 1.03
    elif not pd.isna(row["eqMagMfa"]):
        return 0.67 * row["eqMagMfa"] + 2.07
    elif not pd.isna(row["eqMagnitude"]):
        return row["eqMagnitude"]  # Si no hay otra, usar la general
    else:
        return np.nan

earthquakes["magnitude_Mw"] = earthquakes.apply(to_mw, axis=1)

earthquakes= earthquakes.dropna(subset=["magnitude_Mw"])

# Creamos también una columna donde nos indique la escala de magnitud de donde partíamos, si hay algún dato nulo, elimino esa fila.
def get_mag_source(row):
    if pd.notnull(row["eqMagMw"]): return "Mw"
    elif pd.notnull(row["eqMagMs"]): return "Ms→Mw"
    elif pd.notnull(row["eqMagMb"]): return "Mb→Mw"
    elif pd.notnull(row["eqMagMl"]): return "ML"
    return "None"

earthquakes["MagnitudeSource"] = earthquakes.apply(get_mag_source, axis=1)
earthquakes= earthquakes.dropna(subset=["MagnitudeSource"])

# Elimino las columnas iniciales de magnitudes, ya que con las dos columnas nuevas tenemos los datos que necesitamos.

mag_columns = [col for col in earthquakes.columns if col.startswith("eqMag") or col == "eqMagnitude"]

earthquakes = earthquakes.drop(columns=mag_columns)


In [15]:
earthquakes.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4806 entries, 0 to 6543
Data columns (total 44 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   id                               4806 non-null   int64  
 1   year                             4806 non-null   int64  
 2   locationName                     4806 non-null   object 
 3   latitude                         4806 non-null   float64
 4   longitude                        4806 non-null   float64
 5   damageAmountOrder                3670 non-null   float64
 6   publish                          4806 non-null   bool   
 7   country                          4806 non-null   object 
 8   regionCode                       4806 non-null   int64  
 9   intensity                        2436 non-null   float64
 10  deathsAmountOrder                2194 non-null   float64
 11  tsunamiEventId                   1582 non-null   float64
 12  deathsAmountOrderTotal   

*Hemos pasado de un dataset con 6544 registros a uno con 4806*

## 3.- FUSIÓN DE LOS DOS DATASETS

Antes de seguir con el tratamiento de datos faltantes, unimos los dos datasets.

Nos quedamos con las siguientes colunmas del dataset de tsunamis (para rellenar huecos en los datos de los terremotos y con interés informativo aunque podría ser interesante para futuros estudios de protección frente a tsunamis):
* 'earthquakeEventId'   
* 'oceanicTsunami'  
* 'maxWaterHeight'  
* 'eqDepth',
* 'deathsAmountOrderTotal',
* 'housesDestroyedTotal',
* 'damageMillionsDollarsTotal'
* 'regionCode'
* 'latitude'
* 'longitude'


Unimos los dos datasets mediante el id de los terremotos.

Queremos estudiar si a partir de un terremoto, tuvo ocurrencia un tsunami, así que los datos faltantes en oceanicTsunami, los imputamos como 0 (no tsunamis) y convertimos esta columna en binaria (será nuestro target)

In [16]:
tsunamis.columns

Index(['id', 'year', 'eventValidity', 'causeCode', 'earthquakeEventId',
       'numDeposits', 'country', 'locationName', 'regionCode', 'latitude',
       'longitude', 'numRunups', 'tsIntensity', 'deathsAmountOrder',
       'damageAmountOrder', 'deathsAmountOrderTotal', 'damageAmountOrderTotal',
       'publish', 'oceanicTsunami', 'volcanoEventId', 'maxWaterHeight',
       'eqMagnitude', 'month', 'housesDestroyedAmountOrder', 'deathsTotal',
       'housesDestroyedAmountOrderTotal', 'day', 'tsMtIi', 'deaths', 'eqDepth',
       'housesDamagedAmountOrder', 'housesDamagedAmountOrderTotal',
       'housesDestroyed', 'housesDestroyedTotal', 'hour', 'minute', 'area',
       'injuries', 'injuriesAmountOrder', 'injuriesTotal',
       'injuriesAmountOrderTotal', 'housesDamaged', 'housesDamagedTotal',
       'second', 'missingTotal', 'missingAmountOrderTotal',
       'damageMillionsDollarsTotal', 'tsMtAbe', 'damageMillionsDollars',
       'warningStatusId', 'missing', 'missingAmountOrder'],
      

In [17]:
colum_ts = [
    'earthquakeEventId', 
    'oceanicTsunami', 
    'maxWaterHeight', 
    'eqDepth',
    'deathsAmountOrderTotal',
    'housesDestroyedTotal',
    'damageMillionsDollarsTotal',
    'regionCode', 
    'latitude',
    'longitude'
]

dfmerge = pd.merge(earthquakes,tsunamis[colum_ts],left_on='id',right_on='earthquakeEventId',how='left')

# NaN en oceanicTsunami → 0 (no tsunami)
dfmerge['oceanicTsunami'] = dfmerge['oceanicTsunami'].fillna(0).astype(int)



In [18]:
dfmerge.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4806 entries, 0 to 4805
Data columns (total 54 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   id                               4806 non-null   int64  
 1   year                             4806 non-null   int64  
 2   locationName                     4806 non-null   object 
 3   latitude_x                       4806 non-null   float64
 4   longitude_x                      4806 non-null   float64
 5   damageAmountOrder                3670 non-null   float64
 6   publish                          4806 non-null   bool   
 7   country                          4806 non-null   object 
 8   regionCode_x                     4806 non-null   int64  
 9   intensity                        2436 non-null   float64
 10  deathsAmountOrder                2194 non-null   float64
 11  tsunamiEventId                   1582 non-null   float64
 12  deathsAmountOrderTot

In [19]:
#Voy a rellenar los datos faltantes del dataset de terremotos con las columnas repetidas seleccionadas en el dataset de tsunamis.
#Eliminaré las filas del dataset resultante en las que eqDepth siga sin tener dato.

cols_repetidas = [
    'eqDepth',
    'deathsAmountOrderTotal',
    'housesDestroyedTotal',
    'damageMillionsDollarsTotal'
]

for col in cols_repetidas:
    col_x = f"{col}_x"
    col_y = f"{col}_y"
    
    if col_x in dfmerge.columns and col_y in dfmerge.columns:
        dfmerge[col_x] = dfmerge[col_x].fillna(dfmerge[col_y])
        dfmerge = dfmerge.rename(columns={col_x: col})
        dfmerge = dfmerge.drop(columns=[col_y])

dfmerge = dfmerge.dropna(subset=['eqDepth']).reset_index(drop=True)

dfmerge[cols_repetidas].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3515 entries, 0 to 3514
Data columns (total 4 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   eqDepth                     3515 non-null   float64
 1   deathsAmountOrderTotal      1508 non-null   float64
 2   housesDestroyedTotal        703 non-null    float64
 3   damageMillionsDollarsTotal  609 non-null    float64
dtypes: float64(4)
memory usage: 110.0 KB


In [20]:
dfmerge

id  year                  locationName  latitude_x  longitude_x  \
0         3 -2000              TURKMENISTAN:  W      38.000       58.200   
1        25  -250        KYRGYZSTAN:  CHIGUCHIN      42.700       77.500   
2        28  -227   GREECE:  DODECANESE ISLANDS      36.300       28.300   
3        38    10           TURKMENISTAN:  NISA      38.000       58.300   
4        55   -63           RUSSIA:  PANTIKAPEY      45.200       36.600   
...     ...   ...                           ...         ...          ...   
3510  10923  2025  RUSSIA:  KAMCHATKA PENINSULA      52.512      160.324   
3511  10924  2025             TURKEY: BALIKESIR      39.312       28.069   
3512  10925  2025         EL SALVADOR-GUATEMALA      14.055      -89.883   
3513  10926  2025   INDONESIA:  SULAWESI:  POSO      -1.282      120.727   
3514  10927  2025          CHILE: DRAKE PASSAGE     -60.186      -61.821   

      damageAmountOrder  publish       country  regionCode_x  intensity  \
0                   1.0     True  TURKMENISTAN            40       10.0   
1                   3.0     True    KYRGYZSTAN            40        9.0   
2                   1.0     True        GREECE           130       10.0   
3                   3.0     True  TURKMENISTAN            40       10.0   
4                   3.0     True        RUSSIA            40        8.0   
...                 ...      ...           ...           ...        ...   
3510                NaN     True        RUSSIA            50        8.0   
3511                2.0     True        TURKEY           140        NaN   
3512                3.0     True     GUATEMALA           100        NaN   
3513                2.0     True     INDONESIA           170        NaN   
3514                NaN     True    ANTARCTICA            20        NaN   

      deathsAmountOrder  tsunamiEventId  deathsAmountOrderTotal  eqDepth  \
0                   1.0             NaN                     1.0     18.0   
1                   NaN             NaN                     NaN     20.0   
2                   1.0          3092.0                     1.0    100.0   
3                   NaN             NaN                     NaN     18.0   
4                   NaN             NaN                     NaN     20.0   
...                 ...             ...                     ...      ...   
3510                NaN          6044.0                     1.0     35.0   
3511                1.0             NaN                     1.0     10.0   
3512                1.0             NaN                     1.0     10.0   
3513                1.0          6046.0                     1.0      8.0   
3514                NaN          6047.0                     NaN     10.0   

      deaths  housesDestroyedAmountOrder  deathsTotal  damageAmountOrderTotal  \
0        1.0                         1.0          1.0                     1.0   
1        NaN                         NaN          NaN                     3.0   
2        NaN                         1.0          NaN                     1.0   
3        NaN                         NaN          NaN                     NaN   
4        NaN                         NaN          NaN                     3.0   
...      ...                         ...          ...                     ...   
3510     NaN                         NaN          1.0                     2.0   
3511     1.0                         1.0          1.0                     2.0   
3512     1.0                         NaN          1.0                     3.0   
3513     1.0                         1.0          1.0                     2.0   
3514     NaN                         NaN          NaN                     NaN   

      housesDestroyedAmountOrderTotal  month   day  volcanoEventId  \
0                                 1.0    NaN   NaN             NaN   
1                                 NaN    NaN   NaN             NaN   
2                                 1.0    NaN   NaN             NaN   
3                            

In [21]:
dfregion=dfmerge

In [22]:
dfregion['same_region'] = dfregion['regionCode_x'] == dfregion['regionCode_y']

print(dfregion['same_region'].value_counts())

same_region
False    3431
True       84
Name: count, dtype: int64


In [23]:
dfmerge.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3515 entries, 0 to 3514
Data columns (total 51 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   id                               3515 non-null   int64  
 1   year                             3515 non-null   int64  
 2   locationName                     3515 non-null   object 
 3   latitude_x                       3515 non-null   float64
 4   longitude_x                      3515 non-null   float64
 5   damageAmountOrder                2672 non-null   float64
 6   publish                          3515 non-null   bool   
 7   country                          3515 non-null   object 
 8   regionCode_x                     3515 non-null   int64  
 9   intensity                        1626 non-null   float64
 10  deathsAmountOrder                1524 non-null   float64
 11  tsunamiEventId                   1070 non-null   float64
 12  deathsAmountOrderTot

In [24]:
dfmerge.isnull().sum()

id                                    0
year                                  0
locationName                          0
latitude_x                            0
longitude_x                           0
damageAmountOrder                   843
publish                               0
country                               0
regionCode_x                          0
intensity                          1889
deathsAmountOrder                  1991
tsunamiEventId                     2445
deathsAmountOrderTotal             2007
eqDepth                               0
deaths                             2073
housesDestroyedAmountOrder         2397
deathsTotal                        2068
damageAmountOrderTotal             1039
housesDestroyedAmountOrderTotal    2365
month                                28
day                                  36
volcanoEventId                     3484
housesDamagedAmountOrder           2557
injuries                           2196
injuriesAmountOrder                2027


In [25]:
dfmerge['oceanicTsunami'].value_counts()

#coinciden los nulos de las longitudes y las latitudes de los tsunamis con que realmente ocurrieran dichos tsunamis

oceanicTsunami
0    2466
1    1049
Name: count, dtype: int64

In [26]:
dfmerge.rename(columns=lambda x: x.replace('_x', '_eq').replace('_y', '_ts'), inplace=True)

In [27]:
dfmerge.columns

Index(['id', 'year', 'locationName', 'latitude_eq', 'longitude_eq',
       'damageAmountOrder', 'publish', 'country', 'regionCode_eq', 'intensity',
       'deathsAmountOrder', 'tsunamiEventId', 'deathsAmountOrderTotal',
       'eqDepth', 'deaths', 'housesDestroyedAmountOrder', 'deathsTotal',
       'damageAmountOrderTotal', 'housesDestroyedAmountOrderTotal', 'month',
       'day', 'volcanoEventId', 'housesDamagedAmountOrder', 'injuries',
       'injuriesAmountOrder', 'injuriesTotal', 'injuriesAmountOrderTotal',
       'hour', 'housesDamagedAmountOrderTotal', 'minute', 'housesDestroyed',
       'housesDamaged', 'housesDestroyedTotal', 'housesDamagedTotal', 'area',
       'second', 'damageMillionsDollars', 'missing', 'missingAmountOrder',
       'missingTotal', 'missingAmountOrderTotal', 'damageMillionsDollarsTotal',
       'magnitude_Mw', 'MagnitudeSource', 'earthquakeEventId',
       'oceanicTsunami', 'maxWaterHeight', 'regionCode_ts', 'latitude_ts',
       'longitude_ts', 'same_region

## 4.- DATASET PROCESADO

Guardamos el dataset procesado. En el siguiente notebook empezaremos con el EDA y  terminaremos de procesar los datos nulos. Al finalizar, dividiremos entre train y test.

Nuestro dataset procesado tiene 3515 registros y 47 columnas.

In [28]:
dfmerge.to_csv("processed.csv", index=False, encoding="utf-8")